sqlmesh/core/scheduler.py

```diff
@@ -431,6 +438,37 @@ def compute_interval_params(
         ignore_cron=ignore_cron,
         end_bounded=end_bounded,
     ).items():
+
+        def intersect(left: SnapshotInterval, intervals: list[Interval]) -> SnapshotInterval | None:
+            """Returns the intersection of `left` against `intervals`. Returns None if it does not intersect. Assumes `intervals` are disjoint."""
+
+            left_lower, left_upper = left
+            for right in intervals:
+                right_lower, right_upper = to_timestamp(right[0]), to_timestamp(right[1])
```

Comment:
how exactly is the api supposed to work for this? i think users should probably be expected to return discrete intervals that align to the interval unit.

Reply:
This was addressed in the new interface in https://github.com/TobikoData/sqlmesh/pull/2822/commits/e9d9119249d07679bc4c04c913d4563ffdf8aef7